src/accelerate/utils/operations.py

```diff
@@ -287,9 +287,26 @@ def _gpu_gather_one(tensor):
         # Can only gather contiguous tensors
         if not tensor.is_contiguous():
             tensor = tensor.contiguous()
-        output_tensors = [torch.empty_like(tensor) for _ in range(torch.distributed.get_world_size())]
-        torch.distributed.all_gather(output_tensors, tensor)
-        return torch.cat(output_tensors, dim=0)
+
+        state = PartialState()
+
+        if state.backend is not None and state.backend != "gloo":
+            output_tensors = torch.zeros(
+                state.num_processes * tensor.numel(),
+                dtype=tensor.dtype,
+                device=state.device,
+            )
+            if is_torch_version(">=", "1.13"):
+                torch.distributed.all_gather_into_tensor(output_tensors, tensor)
+            else:
+                torch.distributed._all_gather_base(output_tensors, tensor)
+            return output_tensors.view(-1, *tensor.size()[1:])
+        else:
+            # gloo does not support `all_gather_into_tensor`, which will result
+            # in a larger memory overhead for the op
+            output_tensors = [torch.zeros_like(tensor) for _ in range(state.num_processes)]
```

Comment:
Again, why `zeros` and not `empty`?

Reply:
Reverted this back to `empty_like` since it's different.